workflows/rnaseq/main.nf

```diff
@@ -782,6 +782,71 @@ workflow RNASEQ {
+    // UMITOOLS_PREPAREFORSALMON.out.log       >> 'star_salmon/umitools/log/intermeds/'
+
+    // TODO: !params.skip_alignment && params.aligner == 'star_salmon'
+    DESEQ2_QC_STAR_SALMON.out.rdata         >> 'star_salmon/deseq2_qc/'
+    DESEQ2_QC_STAR_SALMON.out.pca_txt       >> 'star_salmon/deseq2_qc/'
+    DESEQ2_QC_STAR_SALMON.out.pdf           >> 'star_salmon/deseq2_qc/'
+    DESEQ2_QC_STAR_SALMON.out.dists_txt     >> 'star_salmon/deseq2_qc/'
+    DESEQ2_QC_STAR_SALMON.out.size_factors  >> 'star_salmon/deseq2_qc/'
+    DESEQ2_QC_STAR_SALMON.out.log           >> 'star_salmon/deseq2_qc/'
+
+    // TODO: !params.skip_alignment && params.aligner == 'star_rsem'
+    // DESEQ2_QC_RSEM.out.rdata                >> 'star_rsem/deseq2_qc/'
+    // DESEQ2_QC_RSEM.out.pca_txt              >> 'star_rsem/deseq2_qc/'
+    // DESEQ2_QC_RSEM.out.pdf                  >> 'star_rsem/deseq2_qc/'
+    // DESEQ2_QC_RSEM.out.dists_txt            >> 'star_rsem/deseq2_qc/'
+    // DESEQ2_QC_RSEM.out.size_factors         >> 'star_rsem/deseq2_qc/'
+    // DESEQ2_QC_RSEM.out.log                  >> 'star_rsem/deseq2_qc/'
+
+    QUANTIFY_PSEUDO_ALIGNMENT.out.tpm_gene                      >> "${params.pseudo_aligner}/"
+    QUANTIFY_PSEUDO_ALIGNMENT.out.counts_gene                   >> "${params.pseudo_aligner}/"
+    QUANTIFY_PSEUDO_ALIGNMENT.out.lengths_gene                  >> "${params.pseudo_aligner}/"
+    QUANTIFY_PSEUDO_ALIGNMENT.out.counts_gene_length_scaled     >> "${params.pseudo_aligner}/"
+    QUANTIFY_PSEUDO_ALIGNMENT.out.counts_gene_scaled            >> "${params.pseudo_aligner}/"
+    QUANTIFY_PSEUDO_ALIGNMENT.out.tpm_transcript                >> "${params.pseudo_aligner}/"
+    QUANTIFY_PSEUDO_ALIGNMENT.out.counts_transcript             >> "${params.pseudo_aligner}/"
+    QUANTIFY_PSEUDO_ALIGNMENT.out.lengths_transcript            >> "${params.pseudo_aligner}/"
+    QUANTIFY_PSEUDO_ALIGNMENT.out.merged_gene_rds               >> "${params.pseudo_aligner}/"
+    QUANTIFY_PSEUDO_ALIGNMENT.out.merged_gene_rds_length_scaled >> "${params.pseudo_aligner}/"
+    QUANTIFY_PSEUDO_ALIGNMENT.out.merged_gene_rds_scaled        >> "${params.pseudo_aligner}/"
+    QUANTIFY_PSEUDO_ALIGNMENT.out.merged_transcript_rds         >> "${params.pseudo_aligner}/"
+    // TODO: process 'QUANTIFY_PSEUDO_ALIGNMENT:CUSTOM_TX2GENE'
+
+    DESEQ2_QC_PSEUDO.out.rdata          >> "${params.pseudo_aligner}/deseq2_qc/"
+    DESEQ2_QC_PSEUDO.out.pca_txt        >> "${params.pseudo_aligner}/deseq2_qc/"
+    DESEQ2_QC_PSEUDO.out.pdf            >> "${params.pseudo_aligner}/deseq2_qc/"
+    DESEQ2_QC_PSEUDO.out.dists_txt      >> "${params.pseudo_aligner}/deseq2_qc/"
+    DESEQ2_QC_PSEUDO.out.size_factors   >> "${params.pseudo_aligner}/deseq2_qc/"
+    DESEQ2_QC_PSEUDO.out.log            >> "${params.pseudo_aligner}/deseq2_qc/"
+
+    BBMAP_BBSPLIT.out.stats     >> 'bbsplit/'
+    BBMAP_BBSPLIT.out.all_fastq >> 'bbsplit/intermeds/'
```

Comment:
These mappings are here and not in the bbsplit module because I believe it is used multiple times in the pipeline and in some instances publishing is disabled.

We could move these rules to the module and make it so that re-mapping them to `null` in a workflow essentially disables publishing for that process instance:
```groovy
    BBMAP_BBSPLIT.out.stats     >> null
    BBMAP_BBSPLIT.out.all_fastq >> null
```

Then you could also use the param switches here instead of the output definition:
```groovy
    BBMAP_BBSPLIT.out.all_fastq >> params.save_bbsplit_reads ? 'bbsplit/intermeds/' : null
```